include/coap3/coap_mutex_internal.h

```diff
@@ -50,7 +52,8 @@ typedef mutex_t coap_mutex_t;
 #if NO_SYS
 /* Single threaded, no-op'd in lwip/sys.h */
 typedef int coap_mutex_t;
-#define COAP_MUTEX_INITIALIZER 0
+#define COAP_MUTEX_DEFINE(_name)			\
+	static coap_mutex_t _name
```

Comment:
So that it is properly initialized, I think I would like to see this as
```
    static coap_mutex_t _name = 0
````
even though it does not really make any difference.

Reply:
Since it is `static`, it will be automatically initialized to 0. I think that some compilers (some gcc version really, I don't use anything else) even generate a warning when a `static` is "unnecessary" (aka explicitly) initialized to 0. This is true for sure when `static` variable is defined outside of function body, but not sure about defining it inside function body (in that case there might be no warning). But since this macro could potentially be used outside of function body in the future, I think it would be better to **not** initialize it.

Do you still want to explicitly initialize to 0 in that case? :)

Reply:
Well, above (warning generation by gcc) it true for `int` type, it might be different for `typedef` such as `coap_mutex_t`.

Reply:
Need to checkout for a few hours, will look at it when I get back

Reply:
Agreed, this initialization should not be done.

Reply:
Agreed, this initialization should not be done.